packages/x-telemetry/src/runtime/settings.ts

```diff
@@ -0,0 +1,15 @@
+import { setTelemetryEnvConfigValue } from './config';
+
+const muiXTelemetrySettings = {
+  enableDebug: () => {
+    setTelemetryEnvConfigValue('DEBUG', true);
+  },
+  enableTelemetry: () => {
+    setTelemetryEnvConfigValue('IS_COLLECTING', true);
+  },
+  disableTelemetry: () => {
+    setTelemetryEnvConfigValue('IS_COLLECTING', false);
+  },
+};
```

Comment:
```suggestion
const muiXTelemetrySettings =
  process.env.NODE_ENV === 'production'
    ? { enableDebug: noop, enableTelemetry: noop, disableTelemetry: noop }
    : {
        enableDebug: () => {
          setTelemetryEnvConfigValue('DEBUG', true);
        },
        enableTelemetry: () => {
          setTelemetryEnvConfigValue('IS_COLLECTING', true);
        },
        disableTelemetry: () => {
          setTelemetryEnvConfigValue('IS_COLLECTING', false);
        },
      };
```

To cut unused imports in production as early as possible